### DISCUSSION TITLE
What counts as a "Aircraft destroyed"? (Maybe "bug")

### DISCUSSION THREAD

**Discussion by TrakDah** [Khopa/dcs_liberation#2764] [Q&A] [answered]
So I'm new here, and I have just started playing around with Liberation. 

But one thing that I have noticed is a definite oddity regarding what counts as a "destroyed" aircraft. Unless I am doing something wrong...

I am experimenting with the same mission over and over to try to figure out what's what. In it I'm concentrating on 3 flights of 2 MiG19's. 

Two cases:
Case 1: If I or the AI bomb the aircraft while they're on the ground, they don't count as dead and I'm not awarded those as "Aircraft destroyed" for the mission progression. The aircraft are most definitely destroyed however: they are in a million pieces, burning and some are hardly visible from F11 view. The aircraft are noted as 

Case 2: Same mission, but I add a "Explode 1000" trigger to destroy them from the F10 menu. In this case they are validated as "Aircraft destroyed" by Liberty, and of course by DCS too.

Also, just to be specific: If I first do Case 1 and then Case 2 I also get "Aircraft destroyed".

THE PROBLEM:
When I look in DCS, the munitions (I'm testing with BK90 MJ1) have had the desired effect, and DCS's Debriefing states:
Event: Kill --- Details: MJ1

Note "Kill". 

But Liberation doesn't see that as a kill. The reason seems to be that after the "kill" event, the units are still visible from the F10 map. They are killed, but not ... dead?
After I explode them (see case 2) they are in fact removed from the F10 map and that's when Liberation sees them as "dead".
But note that there are no more events in DCS's Debriefing for those units after they've been marked as "kill"; my explosion doesn't show in the debriefing at all. It does _something_, since the units are removed from the F10 map... SuperDead. 

Also, the above is true for any aircraft that's visible on the F10 map at the time of the attack, be they Uncontrolled or "normal".

WHY IS THIS IMPORTANT?
Well, I would very much like to be able to take out enemy aircraft on the ground and have it count. 

So what gives? Any ideas or explanations? 

And forgive me if this has been addressed before; I've searched but haven't found anything.

![case1](https://user-images.githubusercontent.com/25872334/226452409-061d2192-025f-4734-bf0a-57dc379e7562.png)
![case2](https://user-images.githubusercontent.com/25872334/226452427-b6b92cb9-5e7a-40ed-bb78-cd50e58f5236.png)
![debrief](https://user-images.githubusercontent.com/25872334/226452442-52bb6728-5dc1-4a48-99af-3cd42ba70a05.png)

**Comment by Starfire13** [comment]
All Liberation is looking for is the kill event sent by DCS. It is essentially the BDA red text you see on the side of your screen when you kill a target. This is a problem because DCS doesn't actually register a kill event for parked aircraft unless they're actually killed-killed. They can be in three pieces and burning but DCS doesn't consider them killed. Eventually they will despawn. What you need to do to get the kill to register is essentially wipe them off the face of the planet in one hit. Something like a 2000 lb bomb does the trick. Cluster bombs won't do it.  Gun rounds and rockets won't do it. The A-10's cannon can potentially kill a parked aircraft, if you pour almost all your ammo capacity into a single target. Multiple hellfires aren't powerful enough to get a kill either.

**Comment by TrakDah** [comment]
Yes, so you confirm what I saw. 

But that "kill" event must surely be interceptable. I mean, DCS "sees" it (for the debrief). Shouldn't Liberation also try to rather intercept that one event rather than the SuperKill one?

Anyhow, this imho is a bug, as it makes for a huge wrt to how Liberation can be used in so far  that it precludes strategically hitting parked aircraft with anything but ridiculously heavy bombs. And as we know: all you need irl is a hand grenade on a drone.

But the SOLUTION is for Liberation to try intercepting the "kill" event.

**Comment by Starfire13** [comment]
There is no such thing as a superkill event. All kills are logged the same way. From what I've seen so far, there is no actual kill event logged by DCS when you simply break a plane apart and it burns till it despawns. You only get a kill event if you actually kill it (meaning BDA tells you it is killed). DCS writes all kill events to a file (state.json), which Liberation then reads. If you're saying that there is a kill event listed in that file that Liberation is not recognising, then that is potentially a bug. But it is not one that I am encountering.

**Comment by TrakDah** [comment]
Schrödinger's units: dead and not dead at the same time... Killed (in game) but not killed in the json. 
The bug is on the party that writes that json. Who does? It's placed in the Liberation folder, so I doubt it's written by ED. 

![image](https://user-images.githubusercontent.com/25872334/226547785-0ca780d5-2b99-405b-9dbc-b0b10d547bc9.png)

**Comment by TrakDah** [comment, marked as answer]
So this is definitely a bug on Liberation end.
The solution to fix it is to change the behavior in dcs_liberation.lua
In it, it's stated:
-- killed aircraft will be added via S_EVENT_CRASH event

This needs to be replaced with S_EVENT_UNIT_LOST, on line 145.
When I did, the units were marked as "Aircraft Destroyed" when they were "Kill"-marked inside the mission.

I have not tested, but I think this is a fix that should be used for ground units to make it better/more predictable.

![image](https://user-images.githubusercontent.com/25872334/226568672-419e30da-fb08-4ae0-bca5-fec389791510.png)

**Comment by Starfire13** [comment]
Ok.  Please file a bug report for this.
